scm-webapp/src/main/java/sonia/scm/api/v2/resources/RepositoryExportResource.java

```diff
@@ -126,35 +122,49 @@ public Response exportRepository(@Context UriInfo uriInfo,
   }
 
   private Response exportRepository(Repository repository, boolean compressed) {
-    StreamingOutput output = os -> {
-      try (RepositoryService service = serviceFactory.create(repository)) {
-        if (compressed) {
-          GzipCompressorOutputStream gzipCompressorOutputStream = new GzipCompressorOutputStream(os);
-          service.getBundleCommand().bundle(gzipCompressorOutputStream);
-          gzipCompressorOutputStream.finish();
-        } else {
-          service.getBundleCommand().bundle(os);
+    StreamingOutput output;
+    String fileExtension;
+    try (final RepositoryService service = serviceFactory.create(repository)) {
+      BundleCommandBuilder bundleCommand = service.getBundleCommand();
+      fileExtension = resolveFileExtension(bundleCommand, compressed);
+      output = os -> {
+        try {
+          if (compressed) {
+            GzipCompressorOutputStream gzipCompressorOutputStream = new GzipCompressorOutputStream(os);
+            bundleCommand.bundle(gzipCompressorOutputStream);
+            gzipCompressorOutputStream.finish();
+          } else {
+            bundleCommand.bundle(os);
+          }
+        } catch (IOException e) {
+          throw new InternalRepositoryException(repository, "repository export failed", e);
         }
-      } catch (IOException e) {
-        throw new InternalRepositoryException(repository, "repository export failed", e);
-      }
-    };
+      };
+    }
 
     return Response
       .ok(output, compressed ? "application/x-gzip" : MediaType.APPLICATION_OCTET_STREAM)
-      .header("content-disposition", createContentDispositionHeaderValue(repository, compressed))
+      .header("content-disposition", createContentDispositionHeaderValue(repository, fileExtension))
       .build();
   }
 
-  private String createContentDispositionHeaderValue(Repository repository, boolean compressed) {
+  private String resolveFileExtension(BundleCommandBuilder bundleCommand, boolean compressed) {
+    String fileExtension = bundleCommand.getFileExtension();
+    if (compressed) {
+      fileExtension += ".gz";
+    }
+    return fileExtension;
```

Comment:
Peronally, I find this more readable:
```suggestion
    if (compressed) {
      return bundleCommand.getFileExtension() + ".gz";
    } else {
      return bundleCommand.getFileExtension();
    }
```